### ISSUE TITLE
detach log verbose and confusing

### ISSUE THREAD

**Issue by panlinux** [CanonicalLtd/ubuntu-advantage-client#311]
I attached a bionic system, and this was the status output:
```
Account: andreas.hasenack@canonical.com
Subscription: andreas.hasenack@canonical.com
Valid until: 0001-01-01
Technical support level: not included

cis-audit     none             inactive
cc            none             inactive
esm           entitled         n/a
fips          none             inactive
fips-updates  none             inactive
livepatch     entitled         inactive
```
I then detached it:
```
$ sudo ua detach
[INFO]: Removing authenticated apt repo: https://esm.staging.ubuntu.com/
This machine is now detached
```
Two things:
- I didn't have esm enabled, so why is it telling me that it's removing the esm repo?
- the attach operation was silent compared to the detach one. A corresponding INFO message was never shown when I attached

**Comment by panlinux** [comment]
That INFO message upon detach is also echoed to the log in `/var/log/ubuntu-advantage.log`

**Comment by panlinux** [comment]
On trusty, it said it was disabling/removing common criteria:
```
ubuntu@trusty-ua:~$ ua status
Account: andreas.hasenack@canonical.com
Subscription: andreas.hasenack@canonical.com
Valid until: 0001-01-01
Technical support level: not included

cis-audit     none             inactive
cc            entitled         n/a
esm           entitled         inactive
fips          entitled         n/a
fips-updates  none             inactive
livepatch     entitled         active

Enable entitlements with `ua enable <service>`

ubuntu@trusty-ua:~$ sudo ua detach
[INFO]: Removing authenticated apt repo: https://esm.staging.ubuntu.com/cc
Removing packages: ubuntu-commoncriteria
This machine is now detached

```

**Comment by OddBloke** [comment]
This is no longer a problem for ESM because #382 stopped using the code path that emitted this message. The current `ua detach` output looks like this:

```
[INFO]: Removing authenticated apt repo: https://esm.staging.ubuntu.com/cc
Removing packages: ubuntu-commoncriteria
This machine is now detached
```

Those two top lines are what we're looking to clean up in this issue.

**Comment by OddBloke** [comment]
OK, this turns out to be a little tricker than I was hoping. Those lines are emitted in `uaclient.apt.remove_auth_apt_repo`: https://github.com/CanonicalLtd/ubuntu-advantage-client/blob/f9693fa0a8d48eda7ee3419d74309f16aa3adee7/uaclient/apt.py#L156 and in `CommonCriteriaEntitlement.disable`: https://github.com/CanonicalLtd/ubuntu-advantage-client/blob/f9693fa0a8d48eda7ee3419d74309f16aa3adee7/uaclient/entitlements/cc.py#L44

The former is called by the latter, so this boils down to the fact that `CommonCriteriaEntitlement.disable` runs in its entirity, despite the fact that the first thing it does is `return False` if `not self.can_disable()`: https://github.com/CanonicalLtd/ubuntu-advantage-client/blob/f9693fa0a8d48eda7ee3419d74309f16aa3adee7/uaclient/entitlements/cc.py#L25-L31

The relevant check in `BaseEntitlement.can_disable()` that we reach for Common Criteria is: https://github.com/CanonicalLtd/ubuntu-advantage-client/blob/f9693fa0a8d48eda7ee3419d74309f16aa3adee7/uaclient/entitlements/base.py#L64-L69

Note here that we only `return False` from `can_disable()` if the entitlement is inactive. This means that we `return True` if it is either active or, crucially, inapplicable. Common Criteria is only applicable on `xenial` so we will attempt to disable it on `trusty` or `bionic` (which matches with the reports in this issue).

The obvious fix here is to modify `BaseEntitlement.can_disable` to `return False` if an entitlement is inapplicable. Unfortunately, I think there's a corner case that will be triggered if we do that: if an entitlement becomes inapplicable after it is enabled, then users will no longer be able to disable it. A couple of cases where this could happen: a user enables livepatch when on a livepatchable kernel, but then boots to an unlivepatchable kernel (e.g. they move from trusty HWE to trusty GA), or a user enables ESM on trusty but then upgrades their machine to xenial.

To summarise, we attempt to disable CC because `BaseEntitlement.can_disable` returns `True` if an entitlement is inapplicable, but if we make that return `False` in that case instead, users won't be able to disable entitlements that are no longer applicable.

My feeling is that we should make that trade-off and then, as a separate piece of work, resolve how to appropriately support the aforementioned corner case. What do other people think?

---

(As a side note, one solution I considered was changing the calls to `.disable()` during detach to be quiet, but that isn't correct; we _do_ want output from entitlements that are legitimately being disabled.)

**Comment by powersj** [comment]
> My feeling is that we should make that trade-off and then, as a separate piece of work, resolve how to appropriately support the aforementioned corner case. What do other people think?

+1 and very well said, well done finding it

Also I am also in agreement with your very last statement that we should output from entitlements that are legitimately being disabled.
